src/sync/partial_sync.cpp

```diff
@@ -58,6 +60,81 @@ void update_schema(Group& group, Property matches_property)
 
 } // unnamed namespace
 
+std::string get_default_name(Query query) {
+    // Include table name as part of key to be able to disambiguate the same query on different
+    // tables.
+    std::string object_class = query.get_table()->get_name().substr(6);
+    return util::format("[%1] %2", object_class, query.get_description());
+}
+
+void register_query(Group& group,
+                    std::string const& key,
+                    std::string const& object_class,
+                    std::string const& serialized_query,
+                    _impl::ResultsNotifier& notifier)
+{
+    auto table = ObjectStore::table_for_object_type(group, "__ResultSets").get();
```

Comment:
Is there a reason why you opted to reimplement the logic in this function rather than reusing the very similar existing logic in the existing `register_query` implementation?